### PR TITLE
SQL: Only resolve aliases if asked for local cluster

### LINKED ISSUE #81157: SQL: SYS TABLES returns local aliases for remote clusters
A query like `SYS TABLES CATALOG LIKE 'remoteCluster' LIKE '%' TYPE 'VIEW'` will return the local aliases. Expected is to return nothing (as the remote aliases are not discoverable through field caps).

### PR DESCRIPTION
**Pull request by bpintea**
This fixes an issue where the local aliases are returned also in the
case the cluster patern indicates a remote cluster. The alias resolution
is now guarded by a cluster pattern match.

Fixes #81157

### CODE REVIEW

**Review by Luegg — APPROVED** [review]
Lgtm but I couldn't find any reference for this syntax in the docs. Is this feature not documented?

File: x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java [hunk 1]
```diff
@@ -216,7 +216,7 @@ public void resolveNames(
 
         String[] indexWildcards = Strings.commaDelimitedListToStringArray(indexWildcard);
         Set<IndexInfo> indexInfos = new HashSet<>();
-        if (retrieveAliases) {
+        if (retrieveAliases && (clusterWildcard == null || simpleMatch(clusterWildcard, clusterName))) {
```
Comment (astefan) [inline x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java:219]: This check - `clusterWildcard == null || simpleMatch(clusterWildcard, clusterName)` - is being used in another method in the same class. Maybe extract it as a separate method?

**Review by astefan — APPROVED** [review]
LGTM

File: x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java (hunk 1 above)
Comment (bpintea) [reply x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java]: Makes sense, extracted.

**Comment by bpintea** [comment]
> couldn't find any reference for this syntax in the docs. Is this feature not documented?

It's not, will open a PR.

**Pull request merged**